File: edge-cases/java/try-with-resources/Sample.java
package edge.resources;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.Reader;

public final class Sample {
    public String firstLine(Reader input) throws IOException {
        try (BufferedReader reader = new BufferedReader(input)) {
            return reader.readLine();
        }
    }
}

File: fixtures/edge-static-j2k/try-with-resources/Sample.kt
package edge.resources

import java.io.BufferedReader
import java.io.IOException
import java.io.Reader

class Sample {
    @Throws(IOException::class)
    fun firstLine(input: Reader): String {
        BufferedReader(input).use { reader ->
            return reader.readLine()
        }
    }
}
